src/repr/src/lib.rs

```diff
@@ -51,9 +57,322 @@ pub use scalar::{
     arb_datum, AsColumnType, Datum, DatumType, PropArray, PropDatum, PropDict, PropList,
     ProtoScalarType, ScalarBaseType, ScalarType,
 };
+use serde::{Deserialize, Serialize, Serializer};
 
 // Concrete types used throughout Materialize for the generic parameters in Timely/Differential Dataflow.
 /// System-wide timestamp type.
-pub type Timestamp = u64;
+#[derive(
+    Clone,
+    // TODO: De-implement Copy, which is widely used.
+    Copy,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Default,
+    Arbitrary,
+)]
+pub struct Timestamp {
+    /// note no `pub`.
+    internal: u64,
+}
```

Comment:
I feel like a large hunk of the implementations that follow could be `Deref<Target = u64>`. That .. still locks us in to more `u64` than I'd like, but not obviously any differently than to and from `[u8; 8]`. I'm not sure what the longer term evolution plans are for the type, and whether we should be moving away from these sorts of methods, or whether we are stuck and make them as simple as possible.

Reply:
The only thing we want to do at some point is change to i64. I don't have an opinion on the Deref stuff, but if someone feels strongly I/they can change to that later.